lib/main.dart

```diff
@@ -15,67 +15,92 @@ void main() {
   runApp(const App());
 }
 
-// In the future, this will be loaded from a proper source rather than being hardcoded.
-final config = Config.ofNetworks([
-  const Network(
-    name: NetworkName.testnet,
-    walletProxyConfig: WalletProxyConfig(
-      baseUrl: 'https://wallet-proxy.testnet.concordium.com',
+/// Load fundamental configuration from the source of truth.
+Future<Config> loadConfig(HttpService http) async {
+  // In the future, this will be loaded from a proper source rather than being hardcoded.
+  return Config.ofNetworks([
+    const Network(
+      name: NetworkName.testnet,
+      walletProxyConfig: WalletProxyConfig(
+        baseUrl: 'https://wallet-proxy.testnet.concordium.com',
+      ),
     ),
-  ),
-]);
+  ]);
+}
+
+Future<ServiceRepository> bootstrap() async {
+  const http = HttpService();
+  final config = await loadConfig(http);
+  final prefs = await SharedPreferences.getInstance();
```

Comment:
```suggestion
  final configFuture = loadConfig(http);
  final prefsFuture = SharedPreferences.getInstance();
  final config = await configFuture;
  final prefs = await prefsFuture;
```

I looked into `Future.wait` but then the return type of the isolates needs to be the same. In this case they would be cast to `Object` to a type cast would be needed below.

Reply:
Nice :+1: